File: app/src/main/java/com/guideapp/guideapp/ui/views/local/LocalContract.java
package com.guideapp.guideapp.ui.views.local;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.app.LoaderManager;
import android.widget.ImageView;

import com.guideapp.guideapp.model.Local;

import java.util.List;

interface LocalContract {
    interface View {
        void showLocals(List<Local> locals);

        void showLocalDetailUi(Local local, ImageView view);

        void showProgressBar();

        void hideProgressBar();

        Context getContext();
    }

    interface Presenter {
        void loadLocals(LoaderManager loaderManager);

        void restartLoadLocals(LoaderManager loaderManager, long idCategory);

        void openLocalDetails(@NonNull Local local, ImageView view);
    }
}

File: app/src/main/java/com/guideapp/ui/views/local/LocalContract.kt
package com.guideapp.ui.views.local

import android.content.Context
import android.support.v4.app.LoaderManager
import android.widget.ImageView

import com.guideapp.model.Local

internal interface LocalContract {
    interface View {
        fun showLocals(locals: List<Local>)

        fun showLocalDetailUi(local: Local, view: ImageView)

        fun showProgressBar()

        fun hideProgressBar()

        fun getContext(): Context
    }

    interface Presenter {
        fun loadLocals(loaderManager: LoaderManager)

        fun restartLoadLocals(loaderManager: LoaderManager, idCategory: Long)

        fun openLocalDetails(local: Local, view: ImageView)
    }
}
